packages/contracts-bedrock/src/L1/SystemConfig.sol

```diff
@@ -268,28 +271,20 @@ contract SystemConfig is OwnableUpgradeable, ISemver {
     }
 
     /// @notice Internal setter for the gas paying token address, includes validation.
-    ///         Ether uses a default value of 18 decimals and the requirement
-    ///         that the decimals are less than or equal to 18 ensures that no precision is
-    ///         lost for tokens that have less decimals.
     /// @param _token Address of the gas paying token.
     function _setGasPayingToken(address _token) internal {
-        if (_token == address(0)) _token = Constants.ETHER;
-
-        uint8 decimals = 18;
-        bytes32 name = bytes32("Ether");
-        bytes32 symbol = bytes32("ETH");
-        if (_token != Constants.ETHER) {
-            decimals = ERC20(_token).decimals();
-            require(decimals == 18, "SystemConfig: bad decimals of gas paying token");
-            name = GasPayingToken.sanitize(ERC20(_token).name());
-            symbol = GasPayingToken.sanitize(ERC20(_token).symbol());
-        }
-
-        if (_token != Constants.ETHER) {
-            GasPayingToken.set({ _token: _token, _decimals: decimals, _name: name, _symbol: symbol });
+        if (_token != address(0) && _token != Constants.ETHER) {
+            require(
+                ERC20(_token).decimals() == GAS_PAYING_TOKEN_DECIMALS, "SystemConfig: bad decimals of gas paying token"
+            );
+            bytes32 name = GasPayingToken.sanitize(ERC20(_token).name());
+            bytes32 symbol = GasPayingToken.sanitize(ERC20(_token).symbol());
+
+            // Set the gas paying token in storage and in the OptimismPortal.
+            GasPayingToken.set({ _token: _token, _decimals: GAS_PAYING_TOKEN_DECIMALS, _name: name, _symbol: symbol });
             OptimismPortal(payable(optimismPortal())).setGasPayingToken({
                 _token: _token,
-                _decimals: decimals,
+                _decimals: GAS_PAYING_TOKEN_DECIMALS,
```

Comment:
hardcoding 18 could work but also using the constant could work

Reply:
my waterloo professors would be sad if I used a hardcoded magic number instead of a constant :P